libs/api_common/windows_platform_common.cpp

```diff
@@ -217,6 +222,8 @@ get_program_type_windows(const GUID& program_type)
 {
     ebpf_result_t result;
 
+    _load_ebpf_provider_data();
```

Comment:
so now the cost of acquiring a lock and checking if a map is empty distributed to all API calls. Is it worth it?

Reply:
This isn't a hot path, so I am not sure the cost of the lock matters.